tools/container/docker-entrypoint.sh

```diff
@@ -0,0 +1,130 @@
+    exit 1
+fi
+
+icinga2_log "information" "Icinga 2 Docker entrypoint script started."
+
+ca="/var/lib/icinga2/certs/ca.crt"
+if [ ! -e "$ca" ]; then
+    nodeSetup=("node" "setup")
+    runNodeSetup=false
+
+    # The following loop looks for environment variables that start with ICINGA_ and applies some transformations
+    # to the keys before processing them in one way or another. Their values are never modified or printed in
+    # unintended ways. The key transformations have the following rules and are applied in the order they are listed:
+    #
+    # - Since it only processes environment variables that start with ICINGA_, it'll first strip that prefix.
+    #   It then passes the key through awk to convert it to lowercase e.g. ICINGA_CN becomes cn.
+    # - For each key, that hits one of the cases below, it will be processed a bit differently. In the first match,
+    #   the environment variable is expected to be a boolean (1 or 0) and it only becomes part of the node setup
+    #   command if and only if its value is 1. In that case, underscores in the key are replaced with dashes and
+    #   passed as-is to the node setup command (e.g., ICINGA_ACCEPT_COMMANDS=1 becomes --accept-commands).
+    # - For the second match, the key is expected to be a key-value pair that should be passed to the node setup
+    #   command. In this case, the key is transformed in the same way as above, i.e., underscores are replaced with
+    #   dashes and the value is passed as-is (e.g., ICINGA_CN=example.com becomes --cn example.com).
+    # - For the third match, the trusted certificate is expected to be a PEM-encoded certificate that should be
+    #   written to a temporary file and passed to the node setup command.
+    # - Lastly, the CA certificate is likewise expected to be a PEM-encoded certificate that should be written to
+    #   the expected location at /var/lib/icinga2/certs/ca.crt.
+    #
+    # When encountering an environment variable prefixed with ICINGA_ that we don't know how to handle, we log it
+    # as an informational message and continue processing the next environment variable but it doesn't cause the
+    # script to fail.
+    for k in $(env | grep -E '^ICINGA_' | cut -d= -f1 || true); do
+        # Strip the ICINGA_ prefix and convert the key to lowercase.
+        key=$(echo "${k#ICINGA_}" | awk '{print tolower($0)}')
+        # Get the value of the environment variable.
+        value="${!k}"
+
+        case "$key" in
+            "accept_commands" | "accept_config" | "disable_confd" | "master")
+                runNodeSetup=true
+                if [ "$value" = "1" ]; then
+                    nodeSetup+=("--${key//_/-}")
+                fi
+                ;;
+            "cn" | "endpoint" | "global_zones" | "listen" | "parent_host" | "parent_zone" | "zone" | "ticket")
+                runNodeSetup=true
+                nodeSetup+=("--${key//_/-}" "$value")
+                ;;
```

Comment:
This must not replace `_` with `-` here. For reasons only known to the gods of Icinga 2, it's `--parent_host`, `--parent_zone`, and `--global_zones`. And it's not just an alias, it's rejected with a `-`:

```
[2025-07-30 12:37:16 +0000] critical/icinga-app: Error while parsing command-line options: unrecognised option '--global-zones'
```

(Though the case just above has to replace it as it's `--accept-config`, `--accept-commands`, and `--disable-confd`.)

Reply:
I also noticed it while testing https://github.com/Icinga/icinga2/pull/10505#discussion_r2242399278 🙈